sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubServiceClient_extensions.cs

```diff
@@ -0,0 +1,367 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Messaging.WebPubSub
+{
+    /// <summary>
+    /// Azure Web PubSub Service Client.
+    /// </summary>
+    [CodeGenSuppress("WebPubSubServiceClient", typeof(string), typeof(Uri), typeof(WebPubSubServiceClientOptions))]
+    public partial class WebPubSubServiceClient
+    {
+        private AzureKeyCredential credential;
+
+        private const string JsonContent = "application/json";
+
+        /// <summary>
+        /// The hub.
+        /// </summary>
+        public string Hub => hub;
+
+        /// <summary>
+        /// The service endpoint.
+        /// </summary>
+        public Uri Endpoint => endpoint;
+
+        /// <summary> Initializes a new instance of WebPubSubServiceClient. </summary>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="hub"> Target hub name, which should start with alphabetic characters and only contain alpha-numeric characters or underscore. </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        public WebPubSubServiceClient(Uri endpoint, string hub, AzureKeyCredential credential)
+            : this(endpoint, hub, credential, new WebPubSubServiceClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a new instance of WebPubSubServiceClient. </summary>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="hub"> Target hub name, which should start with alphabetic characters and only contain alpha-numeric characters or underscore. </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        public WebPubSubServiceClient(Uri endpoint, string hub, AzureKeyCredential credential, WebPubSubServiceClientOptions options)
+        {
+            if (hub == null)
+            {
+                throw new ArgumentNullException(nameof(hub));
+            }
+            if (credential == null)
+            {
+                throw new ArgumentNullException(nameof(credential));
+            }
+            if (endpoint == null)
+            {
+                throw new ArgumentNullException(nameof(endpoint));
+            }
+
+            this.credential = credential;
+            this.hub = hub;
+            this.endpoint = endpoint;
+
+            options ??= new WebPubSubServiceClientOptions();
+            _clientDiagnostics = new ClientDiagnostics(options);
+            apiVersion = options.Version;
+
+            Pipeline = HttpPipelineBuilder.Build(
+                options,
+                new HttpPipelinePolicy[] { new LowLevelCallbackPolicy() },
```

Comment:
In the health client, this policy was being set per retry but here it's being set per call?  We should definitely add guidance somewhere so this is always placed in the same part of the pipeline.  (It might even be worth adding support in `HttpPipelineBuilder` so we can't not be consistent - @pakrym?)

Reply:
I think it was a recent change to make the callback be per-call https://github.com/Azure/autorest.csharp/pull/1279

Reply:
Can we build that into the `HttpPipelineBuilder` with a `bool includeCallbackPolicy` or similar?

Reply:
Definitely, after we stabilize LowLevelCallbackPolicy and move it into code.